samza-core/src/main/java/org/apache/samza/config/JavaStorageConfig.java

```diff
@@ -96,6 +97,11 @@ public boolean getAccessLogEnabled(String storeName) {
     return getBoolean(String.format(ACCESSLOG_ENABLED, storeName), false);
   }
 
+  // Flag to use hdfs restore vs changelog restore, with changelog as default method
+  public boolean getHDFSRestoreEnabled() {
```

Comment:
Let's talk about this. You also need to disable the changelog write if using HDFS.